src/CoreWCF.WebHttp/src/CoreWCF/Channels/StreamBodyWriter.cs

```diff
@@ -0,0 +1,237 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Xml;
+
+namespace CoreWCF.Channels
+{
+    public abstract class StreamBodyWriter : BodyWriter
+    {
```

Comment:
StreamBodyWriter is a general purpose class so can live in Primitives. BufferedBytesStreamBodyWriter would go with it as it's needed when creating a buffered copy. You would then have ActionOfStreamBodyWriter live in WebHttp and move the internal static CreateStreamBodyWriter to be a factory method on ActionOfStreamBodyWriter. This change enables  BufferManagerOutputStream to remain internal.